0xProject/0x-api
# feat: [rfqm] Get transaction hash and store before submitting

~~**Requires 0xProject/protocol#310 to push before tests will pass**~~

**Closes MKR-96**

- Adds Ethers as a direct dependency of 0x API
- Adds function to sign and hash a transaction using ethers
- Adds function to submit a signed transaction still using web3wrapper
- Adds a Presubmit status to the RFQM Transaction Status enum

Now, when a worker is going to submit a transaction, the flow is:
1. Sign and hash the transaction offline
1. Save the transaction to the rfqm_transaction_submission table with a status of Presubmit
1. Submit the transaction to the mempool
1. Ensure the hash computed offline matches the hash from submission (a unit test is added to verify this works as expected, but the implementation also checks and logs an error if they differ)
1. Update the row in rfqm_transaction_submission with a status of Submitted

## Testing Instructions
Unit test added to ensure that the offline hash is the same as the hash from the rpc provider.

Checklist
 Update documentation as needed. Website Documentation PR:

 Prefix PR title with [WIP] if necessary.

 Add tests to cover changes as needed.

 Test changes on the staging environment with Matcha API Staging.

 SRA/Limit orders
 Swap endpoints
 Meta transaction endpoints
 Depth charts
For more information see 0x API Matcha smoke test runbook in Quip.
